### DISCUSSION TITLE
How can dbt-based asset be updated without redeploying Dagster code location?

### DISCUSSION THREAD

**Discussion by mudravrik** [dagster-io/dagster#21668] [Q&A]
We have an mvp integration between dbt Cloud and Dagster Cloud as described in [documentation](https://docs.dagster.io/integrations/dbt-cloud#step-4-cache-the-dbt-cloud-job-compilation).

Github action is set to compile dbt job artifact with graph updates after every push to main branch so the asset configuration is fresh on this side.
However it looks like Dagster does not pick it up until we redeploy the Code Location where corresponding `load_assets_from_dbt_cloud_job()` call is located.
Is this an intended behaviour or we missed something in configuration of Dagster/assets?

**Comment by maximearmstrong** [comment]
> However it looks like Dagster does not pick it up until we redeploy the Code Location where corresponding `load_assets_from_dbt_cloud_job()` call is located.

@mudravrik Do you mean until you click the `Redeploy` button in the Code Location page? Like [this](https://docs.dagster.io/dagster-plus/managing-deployments/code-locations#redeploying-code-locations)?

**Reply by mudravrik** [reply]
Exactly!
However, you directed me to a solution, I can run following commands from dbt-repo action updating the job artifacts:
```
dagster-cloud workspace pull --user-token token -o org -d prod --location-name loc >> workspace.yaml 
dagster-cloud workspace update-location --user-token token -o org -d prod --location-name loc --from workspace.yaml
```
triggering location update with exactly the same configuration as before.

It looks a bit hacky and I don't want to think what would happen if there is already an update going on in the Dagster deployment ie with code changes, so maybe be there is a better way?

**Comment by mudravrik** [comment]
By the way, in my local Dagster-in-docker instance I can see logs of it polling the dbt job every now and then with following logs:
```
2024-05-08 18:25:49 2024-05-08 11:25:49 +0000 - dagster - INFO - system - Polled run ***. Status: [Success]
```
